File: recursion/recarray.java
import java.util.Scanner;
import java.util.ArrayList;
public class recarray{
    public static Scanner scn=new Scanner(System.in);
    public static void main(String [] args){
        ArrayList <Integer> arr=new ArrayList<>(); 
        input(arr);
        System.out.println(minimum(arr,0));

    }

    public static void input(ArrayList<Integer>arr){
        int n=scn.nextInt();
        for(int i=0;i<n;i++){
            int val=scn.nextInt();
            arr.add(val);
        }
    }
    public static int minimum(ArrayList<Integer>arr,int idx){
        if(arr.size()==idx){
            return Integer.MAX_VALUE;
        }
        int res=minimum(arr,idx+1);
        return Math.min(res,arr.get(idx));
    }
}

File: recursion/recarray.cpp
#include<vector>
#include<iostream>
using namespace std;

/*void display(vector<int>&arr,int idx){
    if(idx==arr.size())
    return;

    cout<<arr[idx]<<" ";
    display(arr,idx+1);

}

int main(){
    vector<int>arr={10,20,30,40,50,60,70};
    display(arr,0);
    return 0;
}
*/
//int max(int);
int display(vector<int>&arr,int idx,int max){
    
    if(idx==arr.size())
    {return max;}
    if(arr[idx]>max){
        max=arr[idx];
    }
    max= display(arr,idx+1,max);
    return max;

}


int main(){
    vector<int>arr={10,20,30,40,50,60,70};
    int res=display(arr,0,0);
    cout<<res;
    return 0;
}

